core/src/epicli/cli/engine/ApplyEngine.py

```diff
@@ -161,6 +169,45 @@ def dry_run(self):
 
         return [*self.configuration_docs, *self.infrastructure_docs]
 
+    def azure_subnet_query_if_required(self):
```

Comment:
This needs to be refactored for sure since its mostly code duplication and on this level we should not interact with the provider correctly. Thats what we build the abstraction layer for.